core/geometry/src/curve/CurveChainWithDistanceIndex.ts

```diff
@@ -220,6 +257,70 @@ export class CurveChainWithDistanceIndex extends CurvePrimitive {
     const c = this._path.clone() as CurveChain;
     return CurveChainWithDistanceIndex.createCapture(c);
   }
+
+  /** Return a deep clone */
+  public override clonePartialCurve(_fractionA: number, _fractionB: number): CurveChainWithDistanceIndex | undefined {
+    if (_fractionA === _fractionB)
+      return undefined;
+    let fractionA = _fractionA;
+    let fractionB = _fractionB;
+    const reversed = _fractionA > _fractionB;
+    if (reversed){
+      fractionA = _fractionB;
+      fractionB = _fractionA;
+    }
+    const chainDistanceA = fractionA * this._totalLength;
+    const chainDistanceB = fractionB * this._totalLength;
+    const newPath = Path.create ();
+    const fragmentA = this.chainDistanceToFragment(chainDistanceA, true);
+    const fragmentB = this.chainDistanceToFragment (chainDistanceB, true);
+    if (!fragmentA || !fragmentB)
+      return undefined;
+    const childCurveIndexA = this._path.childIndex (fragmentA.childCurve);
+    const childCurveIndexB = this._path.childIndex(fragmentB.childCurve);
+    const childFractionA = fragmentA.chainDistanceToAccurateChildFraction (chainDistanceA);
+    const childFractionB = fragmentB.chainDistanceToAccurateChildFraction (chainDistanceB);
+    if (childCurveIndexA === undefined || childCurveIndexB === undefined)
+        return undefined;
+    // add a (possibly reversed) partial clone to newPath
+    const addPartialChild = (childCurve: CurvePrimitive, childFraction0: number, childFraction1: number, reversedClone: boolean): boolean => {
+      if (fractionA === fractionB)
+        return false;
+      let newCurve;
+      if (fractionA === 0.0 && fractionB === 1.0){
+        newCurve = childCurve.clone ();
+        if (reversed)
+          newCurve.reverseInPlace ();
+      } else {
+        newCurve = reversedClone?
+                      childCurve.clonePartialCurve (childFraction1, childFraction0)
+                    : childCurve.clonePartialCurve (childFraction0, childFraction1);
+      }
+    if (newCurve){
+      newPath.children.push(newCurve);
+      return true;
+      }
+    return false;
+    };
+
+    if (fragmentA.childCurve === fragmentB.childCurve){
+      // the two distances are within the same curve.
+      if (addPartialChild (fragmentA.childCurve, childFractionA, childFractionB, reversed))
+        return CurveChainWithDistanceIndex.createCapture (newPath);   // singleton -- children[] does not need to be reversed.
+      return undefined;
+    }
+    addPartialChild (this._path.children[childCurveIndexA], childFractionA, 1.0, reversed);
+    // at least two distinct children are impacted ....
+    for (let childIndex = childCurveIndexA + 1; childIndex < childCurveIndexB; childIndex++){
+      addPartialChild (this._path.children[childIndex].clone (), 0.0, 1.0, reversed);
```

Comment:
Is clone() really needed here? Outside the loop, addPartialChild is invoked without cloned childCurve.